features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp

```diff
@@ -0,0 +1,596 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "UBLOX_AT_CellularStack.h"
+
+using namespace mbed;
+using namespace mbed_cellular_util;
+
+UBLOX_AT_CellularStack::UBLOX_AT_CellularStack(ATHandler &atHandler, int cid, nsapi_ip_stack_t stack_type) : AT_CellularStack(atHandler, cid, stack_type)
+{
+    // URC handlers for sockets
+    _at.set_urc_handler("+UUSORD:", callback(this, &UBLOX_AT_CellularStack::UUSORD_URC));
+    _at.set_urc_handler("+UUSORF:", callback(this, &UBLOX_AT_CellularStack::UUSORF_URC));
+    _at.set_urc_handler("+UUSOCL:", callback(this, &UBLOX_AT_CellularStack::UUSOCL_URC));
+    _at.set_urc_handler("+UUPSDD:", callback(this, &UBLOX_AT_CellularStack::UUPSDD_URC));
+	
+    pendingBytes = 0;
+}
+
+UBLOX_AT_CellularStack::~UBLOX_AT_CellularStack()
+{
+}
+
+nsapi_error_t UBLOX_AT_CellularStack::socket_listen(nsapi_socket_t handle, int backlog)
+{
+    return NSAPI_ERROR_UNSUPPORTED;
+}
+
+nsapi_error_t UBLOX_AT_CellularStack::socket_accept(void *server, void **socket, SocketAddress *addr)
+{
+    return NSAPI_ERROR_UNSUPPORTED;
+}
+
+// Callback for Socket Read URC.
+void UBLOX_AT_CellularStack::UUSORD_URC()
+{
+    int a,b;
+    CellularSocket *socket;
+
+    a =_at.read_int();
+    b =_at.read_int();
+
+    socket = find_socket(a);
+    if (socket != NULL) {
+        socket->rx_avail = true;
+        pendingBytes = b;
+        // No debug prints here as they can affect timing
+        // and cause data loss in UARTSerial
+        if (socket->_cb != NULL) {
+            socket->_cb(socket->_data);
+        }
+    }
+}
+
+// Callback for Socket Read From URC.
+void UBLOX_AT_CellularStack::UUSORF_URC()
+{
+    int a,b;
+    CellularSocket *socket;
+
+    a =_at.read_int();
+    b =_at.read_int();
+
+    socket = find_socket(a);
+    if (socket != NULL) {
+        socket->rx_avail = true;
+        pendingBytes = b;
+        // No debug prints here as they can affect timing
+        // and cause data loss in UARTSerial
+        if (socket->_cb != NULL) {
+            socket->_cb(socket->_data);
+        }
+    }
+}
+
+// Callback for Socket Close URC.
+void UBLOX_AT_CellularStack::UUSOCL_URC()
+{
+    int a;
+    CellularSocket *socket;
+
+    a =_at.read_int();
+    socket = find_socket(a);
+    clear_socket(socket);
+}
+
+// Callback for UUPSDD.
+void UBLOX_AT_CellularStack::UUPSDD_URC()
+{
+    int a;
+    CellularSocket *socket;
+
+    a =_at.read_int();
+    socket = find_socket(a);
+    clear_socket(socket);
+}
+
+
+int UBLOX_AT_CellularStack::get_max_socket_count()
+{
+    return UBLOX_MAX_SOCKET;
+}
+
+int UBLOX_AT_CellularStack::get_max_packet_size()
```

Comment:
If we return `UBLOX_MAX_PACKET_SIZE` here, aren't we telling the user that they can send no more than 1024 bytes per call to `sendto()`/`send()`, when in fact it is possible to send more than that because you "chunk" the data up appropriately?

Reply:
Yes we can send more data now. My idea of not removing that function that user can see that whats the max packet size is and if he want to use that limit so that he can use it. What you think we should remove it as we have handled data more than max packet size.

Reply:
Let's see how ARM reply to https://github.com/ARMmbed/mbed-os/issues/7181.